### DISCUSSION TITLE
About dev environment

### DISCUSSION THREAD

**Discussion by Ashleeey1224** [MarkBind/markbind#2373] [Q&A]
Hi team, I tried to modify codes and try to print output (console.log), but the terminal, debug console... didn't print anything. Any changes I applied didn't run. So I guess probably my dev environment didn't setup properly. That's why I wanna ask for help to check and ensure my environment is correct. 

I followed the steps in the documentation, and all my Prerequisites are updated, I did this
<img width="884" alt="Screenshot 2023-10-20 at 1 41 06 am" src="https://github.com/MarkBind/markbind/assets/105656080/3b96a6f6-95ca-4826-bf23-b6734a6dd483">

When **I re-run "markbin serve",** the terminal prints these: is it okay?
<img width="419" alt="Screenshot 2023-10-20 at 1 42 02 am" src="https://github.com/MarkBind/markbind/assets/105656080/414f2059-83a7-4c6a-8a18-a8f66e0721dd">

Based on my current working issue, I tried to found the ts files to edit: CustomListIconProcessor.ts and NodeProcessor, but nothing change, even the console.log()

**Comment by SPWwj** [comment]
Where do you run the markbin serve command?
You should run `markbin serve` within the Markbind project, for example, in the `markbind\docs` directory. Alternatively, you can create your own Markbind project using `markbind init`.

**Reply by Ashleeey1224** [reply]
Hi, thank you for your reply. For the layout file error, I was running a new project, and it generate a file called "contents", but if I edit any "default" value in the  CustomListIconProcessor.ts, it doesn't change anything. do you have any idea. thxx

**Reply by SPWwj** [reply]
1. You need to run `npm run build:backend` in the markbind `framework project`.
2. Rerun the `markbind serve` in the `markbind website project` (refer to the root directory below, not the content folder)

> I was running a new project, and it generate a file called "contents"
<img width="626" alt="image" src="https://github.com/MarkBind/markbind/assets/30100720/e490d47b-6099-4339-9dc5-752b046a18e7">

**Reply by Ashleeey1224** [reply]
hi thank you again for your reply, where is the "framework project"? I run `npm run build:backend` in the root directory, the terminal prints
<img width="791" alt="Screenshot 2023-10-20 at 10 28 16 am" src="https://github.com/MarkBind/markbind/assets/105656080/ee3364be-c9ee-4b6c-8b1e-a5964237202a">

(I tried to re-clone the project and re-setup dev environment followed the "Setting Up" section

**Reply by Ashleeey1224** [reply]
if possible, after `npm link`, `npm run setup`, what should I do to set up my development environment correctly?

**Reply by SPWwj** [reply]
> where is the "framework project"? 

The `framework` I refer to is the markbind repo.

> npm link, npm run setup, what should I do to set up my development environment correctly?
https://markbind.org/devdocs/devGuide/bootcamp/exploreMarkBind.html has the all the resources for the setting up.


By looking at your screenshop did you edit any file in the markbind repo?
Maybe you could try to run the command in a fresh repo first to see if you have correctly setup your project:
<img width="610" alt="image" src="https://github.com/MarkBind/markbind/assets/30100720/f37edfb5-c570-4c1c-8ca2-587537b86726">

**Reply by Ashleeey1224** [reply]
I redo everything: clone --> `npm link` --> ` npm run setup`  --> `npm run build:backend` (no errors now)
What's the next?  do "Explore MarkBind as a User" this to setup my site as a test site?
If I edit codes and want to test it, I need to run "npm run dev "?

**Reply by SPWwj** [reply]
Yes, You might want to complete the bootcamp tutorial before you begin developing features or editing the framework code. It’s important to familiarize yourself with both user and developer commands.

**Reply by Ashleeey1224** [reply]
Hi sorry, I init a local repo at my pc and download the source code. I was still confused about how to apply code changes and see changes in my test repo. What I did is: create my branch from master, and then edit codes run `npm run build:backend`, and then reload my local repo. Is it correct?
